### PR TITLE
Enabling sanitizers on gitlab runner

### PR DESCRIPTION
**Pull request by sithhell**
This PR enables the undefined and address sanitizer for the gitlab runners.

Builds on #3784.

### CODE REVIEW

File: tests/unit/parallel/algorithms/partition_copy_tests.hpp [hunk 1]
```diff
@@ -390,6 +390,7 @@ void test_partition_copy()
 template <typename IteratorTag>
 void test_partition_copy_exception()
 {
+#if !__has_feature(address_sanitizer)
```
Comment (hkaiser) [inline tests/unit/parallel/algorithms/partition_copy_tests.hpp]: `__has_feature()` might not be defined.

**Comment by msimberg** [comment]
@sithhell would you have time to separate the fixes from the GitLab runner changes?

**Comment by msimberg** [comment]
@sithhell do we still want to do this? We definitely want builds with sanitizers, but I mean the gitlab CI specifically.

**Comment by msimberg** [comment]
@sithhell I'll close this for now. Please reopen if you get the time to work on this again.

**Pull request closed without merging**